es6-lib/decoders/shapefile.js

```diff
@@ -105,6 +105,7 @@ class Shapefile extends Duplex {
 
   _fileGroup(extension, name) {
     name = name ? name.toLowerCase() : 'shapefile';
+    extension = extension ? extension.toLowerCase() : extension;
```

Comment:
would you wanna give this a fallback of `''` just to be a little more explicit?

Reply:
Honestly, I want to avoid making any kind of changes to this functionality beyond lowercasing it if it's there!

Reply:
yeah but what is it if it's not there? null ? undefined ? then you're doing string interpolation with this right below here....so why not empty string?

Reply:
I think that would result in hiding an error (I believe interpolating undefined or nil throws an error?) that would later surface in some other way, in a place that might be harder to discover.

Reply:
There's no nil in javascript. Also interpolating with null or undefined doesn't throw an error. Not a huge deal, so not worth the discussion. I just think the above is a little odd.

Reply:
right you are! I was testing that totally wrong.

Reply:
Ok cool. Also the more I look at this, the weirder it seems. I mean just because extension is truthy doesn't mean it will prevent a runtime error if you call `toLowerCase`. If it is anything but a string, it will be truthy but you will still get a runtime error. I think a better check would be `if (typeof extension === 'string')`